include/mgp_mock.py

```diff
@@ -0,0 +1,1011 @@
+            raise InvalidContextError()
+
+        vertex = _mgp_mock.Vertex(self._edge.start_id, self._edge.graph)
+        return Vertex(vertex)
+
+    @property
+    def to_vertex(self) -> "Vertex":
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        vertex = _mgp_mock.Vertex(self._edge.end_id, self._edge.graph)
+        return Vertex(vertex)
+
+    @property
+    def properties(self) -> Properties:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return Properties(self._edge)
+
+    def __eq__(self, other) -> bool:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        if not isinstance(other, Edge):
+            return NotImplemented
+
+        return self.id == other.id
+
+    def __hash__(self) -> int:
+        return hash(self.id)
+
+
+if sys.version_info >= (3, 5, 2):
+    VertexId = typing.NewType("VertexId", int)
+else:
+    VertexId = int
+
+
+class Vertex:
+    __slots__ = ("_vertex",)
+
+    def __init__(self, vertex):
+        if not isinstance(vertex, _mgp_mock.Vertex):
+            raise TypeError(f"Expected '_mgp_mock.Vertex', got '{type(vertex)}'")
+
+        self._vertex = vertex
+
+    def __deepcopy__(self, memo):
+        # This is the same as the shallow copy, because we want to share the
+        # underlying C struct. Besides, it doesn't make much sense to actually
+        # copy _mgp_mock.Vertex as that is actually a reference to a graph element
+        # and not a proper value.
+        return Vertex(self._vertex)
+
+    def is_valid(self) -> bool:
+        return self._vertex.is_valid()
+
+    def underlying_graph_is_mutable(self) -> bool:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return not nx.is_frozen(self._vertex.graph)
+
+    @property
+    def id(self) -> VertexId:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return self._vertex.id
+
+    @property
+    def labels(self) -> typing.Tuple[Label]:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return tuple(Label(label) for label in self._vertex.labels)
+
+    def add_label(self, label: str) -> None:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return self._vertex.add_label(label)
+
+    def remove_label(self, label: str) -> None:
+        if not self.is_valid():
+            raise InvalidContextError()
+        return self._vertex.remove_label(label)
+
+    @property
+    def properties(self) -> Properties:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return Properties(self._vertex)
+
+    @property
+    def in_edges(self) -> typing.Iterable[Edge]:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        for edge in self._vertex.graph.in_edges(self.id, keys=True):
+            yield Edge(_mgp_mock.Edge(edge, self._vertex.graph))
```

Comment:
Instead of referencing graph, can you implement here self._vertex.in_edges() method?

Reply:
Again, this `mgp_mock` method that doesn’t use `_mgp_mock` behaves consistently with the analogous `mgp` method that uses `_mgp`. Which this in mind, I don’t see why there needs to be a dedicated method in `_mgp_mock`.